packages/plugin-ext/src/plugin/message-registry.ts

```diff
@@ -28,20 +28,51 @@ export class MessageRegistryExt {
     }
 
     showInformationMessage(message: string,
-                           optionsOrFirstItem: MessageOptions | string | MessageItem,
-                           items: string[] | MessageItem[]): PromiseLike<string | MessageItem | undefined> {
-        return this.proxy.$showInformationMessage(message, optionsOrFirstItem, items);
+        optionsOrFirstItem: MessageOptions | string | MessageItem,
+        items: string[] | MessageItem[]): PromiseLike<string | MessageItem | undefined> {
+        return this.showMessage(MainMessageType.Info, message, optionsOrFirstItem, items);
     }
 
     showWarningMessage(message: string,
-                           optionsOrFirstItem: MessageOptions | string | MessageItem,
-                           items: string[] | MessageItem[]): PromiseLike<string | MessageItem | undefined> {
-        return this.proxy.$showWarningMessage(message, optionsOrFirstItem, items);
+        optionsOrFirstItem: MessageOptions | string | MessageItem,
+        items: string[] | MessageItem[]): PromiseLike<string | MessageItem | undefined> {
+        return this.showMessage(MainMessageType.Warning, message, optionsOrFirstItem, items);
     }
 
     showErrorMessage(message: string,
-                           optionsOrFirstItem: MessageOptions | string | MessageItem,
-                           items: string[] | MessageItem[]): PromiseLike<string | MessageItem | undefined> {
-        return this.proxy.$showErrorMessage(message, optionsOrFirstItem, items);
+        optionsOrFirstItem: MessageOptions | string | MessageItem,
+        items: string[] | MessageItem[]): PromiseLike<string | MessageItem | undefined> {
+        return this.showMessage(MainMessageType.Error, message, optionsOrFirstItem, items);
+    }
+
+    protected async showMessage(type: MainMessageType, message: string,
+        optionsOrFirstItem: MessageOptions | string | MessageItem,
+        rest: string[] | MessageItem[]): Promise<string | MessageItem | undefined> {
```

Comment:
It's a very good improvement. I have tested it. Thank you.

But,  showing message without any items is not working at all.

Please, try this one:
https://github.com/eclipse/che-theia-samples/blob/master/samples/hello-world-frontend-plugin/src/hello-world-frontend-plugin.ts

```
theia.window.showInformationMessage('Hello World!');
```

Reply:
@olexii4 good catch! please try again

Reply:
perfect